src/integrations/admin/activation-indexation-integration.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Yoast\WP\SEO\Integrations\Admin;
+
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+use Yoast\WP\SEO\Conditionals\Traits\Admin_Conditional_Trait;
+use Yoast\WP\SEO\Integrations\Cleanup_Integration;
+
+/**
+ * This integration registers a run of the cleanup routine whenever the plugin is activated.
+ */
+class Activation_Indexation_Integration implements Integration_Interface {
+
+	use Admin_Conditional_Trait;
+
+	/**
+	 * Registers the action to register a cleanup routine run after the plugin is activated.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		add_action( 'wpseo_activate', [ $this, 'register_cleanup_routine' ] );
+	}
+
+	/**
+	 * Registers a run of the cleanup routine if this has not happened yet.
+	 *
+	 * @return void
+	 */
+	public function register_cleanup_routine() {
+		if ( ! \wp_next_scheduled( Cleanup_Integration::START_HOOK ) ) {
```

Comment:
I think we should take a page of the `WPSEO_Installation` class and first check if this is a first time install. If it is, lets not add the cleanup unnecessarily.

In order for this to work though, we must tweak our `add_action` in that class and this one, so that we make the `wpseo_activate` hook run first from **Activation_Indexation_Integration** and then from **WPSEO_Installation** to ensure that we properly check if it's a first time install or not